src/kbmod/regionsearch/utilities.py

```diff
@@ -0,0 +1,210 @@
+        self.samplespercluster = samplespercluster
+        # data are not loaded
+        self.data_loaded = False
+        # data are not generated
+        self.data_generated = False
+
+        sections = [
+            ["samples", ["bary_to_target", "observer_to_target", "observation_geolocation", "cluster_id"]],
+            ["clusters", ["clusters", "clusterdistances"]],
+        ]
+
+        # tmp data overrides data. Only tmp gets written. Only data is controlled in git
+        # for dirname in ["tmp", "data"]:
+        # Only remove cache in tmp. This also implies generating data in tmp.
+        if removecache:
+            for section in sections:
+                filename = self.__filename("tmp", basename, section[0], suffix)
+                if os.path.exists(filename):
+                    os.remove(filename)
+            self.generate_and_save(basename, suffix, format, sections)
+        else:
+            for dirname in ["tmp", "data"]:
+                # sections must be grouped together
+                if not self.data_loaded:
+                    data_loaded_count = 0
+                    for section in sections:
+                        filename = self.__filename(dirname, basename, section[0], suffix)
+                        if os.path.exists(filename) and self.read_table(filename, format, list(section[1])):
+                            data_loaded_count += 1
+                        else:
+                            # early out since this is not a complete set of data
+                            break
+                    if data_loaded_count == len(sections):
+                        self.data_loaded = True
+                        # early out since this is a complete set of data
+                        break
+
+            # if data are not loaded, generate them
+            if not self.data_loaded:
+                self.generate_and_save(basename, suffix, format, sections)
+
+        if not self.data_loaded:
+            raise Exception(f"Could not read or generate data")
+
+    def generate_and_save(self, basename, suffix, format, sections):
+        if self._generate_data():
+            if os.path.isdir("tmp"):
+                for section in sections:
+                    filename = self.__filename("tmp", basename, section[0], suffix)
+                    self.write_table(filename, format, section[1])
+            self.data_loaded = True
+            self.data_generated = True
+
+    def __filename(self, dirname, basename, sectionname, suffix):
+        return f"{dirname}/{basename}-{self.seed}-{self.clustercnt}-{self.samplespercluster}-{self.version}-{sectionname}{suffix}"
+
+    def read_table(self, filename: str, format: str, colnames: typing.List[str]):
+        hold_version = self.version
+        try:
+            table = astropy.table.Table.read(filename, format=format)
+            for metakey in table.meta.keys():
+                setattr(self, metakey, table.meta[metakey])
```

Comment:
self.version is updated manually to mark changes in the file format.
It was convenient as I was changing stuff. The alternative is to continually remove stale files. If we decide to keep this code at all then it will help manage changes in the repository (stuff in "data")